src/resolvers/mutations/animes.js

```diff
@@ -1,87 +1,91 @@
 import sharp from "sharp";
 
-export function updateAnime(root, { id, anime }, context) {
-	return needGroup(context, ADMIN).then(() =>
-		transformAnime(id, anime, now(), context.storage).then(anime =>
-			context.db
-				.collection("animes")
-				.updateOne({ _id: id }, { $set: anime })
-				.then(() => id)
-		)
-	);
-}
+class Animes {
+	constructor() {}
+
+	@requireGroup(ADMIN)
+	async updateAnime(root, { id, anime }, context) {
+		anime = await transformAnime(id, anime, now(), context.storage);
+
+		return context.db
+			.collection("animes")
+			.updateOne({ _id: id }, { $set: anime })
+			.then(() => id);
+	}
 
-export function addAnime(root, { anime }, context) {
-	const time = now();
-	const id = toId(anime.names[0]);
-	return needGroup(context, ADMIN).then(() =>
-		transformAnime(id, anime, time, context.storage).then(anime => {
-			anime.posted_date = time;
-			anime._id = id;
-			anime.tags = anime.tags.map(t => new ObjectID(t));
-			anime.authors = anime.authors.map(a => new ObjectID(a));
-			return context.db
-				.collection("animes")
-				.insertOne({
-					...anime,
-					medias: [],
-					seasons: []
-				})
-				.then(({ insertedId }) => insertedId);
-		})
-	);
+	@requireGroup(ADMIN)
+	async addAnime(root, { anime }, context) {
+		const time = now();
+		const id = toId(anime.names[0]);
+
+		anime = await transformAnime(id, anime, time, context.storage);
+		anime.posted_date = time;
+		anime._id = id;
+		anime.tags = anime.tags.map(t => new ObjectID(t));
+		anime.authors = anime.authors.map(a => new ObjectID(a));
+
+		return context.db
+			.collection("animes")
+			.insertOne({
+				...anime,
+				medias: [],
+				seasons: []
+			})
+			.then(({ insertedId }) => insertedId);
+	}
+
+	@requireGroup(ADMIN)
+	async addSeason(root, { anime, season }, context) {
+		const time = now();
+		const seasonNb = season.season;
+
+		delete season.season;
+		season.edit_date = time;
+		season.posted_date = time;
+
+		return context.db
+			.collection("animes")
+			.updateOne(
+				{ _id: anime },
+				{ $push: { seasons: { $each: [season], $position: seasonNb } } }
+			)
+			.then(() => ({ anime, ...season }));
+	}
 }
 
-function transformAnime(id, anime, time, storage) {
-	return Promise.all([
+async function transformAnime(id, anime, time, storage) {
+	const options = {
+		progressive: true,
+		quality: 100,
+		optimiseScans: true
+	};
+
+	const [cover, background] = await Promise.all([
 		anime.cover &&
 			anime.cover.then(cover =>
 				storage.save(
 					`${id}_cover`,
 					cover,
 					sharp()
 						.resize(180, 250)
-						.jpeg({ progressive: true, quality: 100, optimiseScans: true })
+						.jpeg(options)
 				)
 			),
+
 		anime.background &&
 			anime.background.then(background =>
-				storage.save(
-					`${id}_background`,
-					background,
-					sharp().jpeg({ progressive: true, quality: 100, optimiseScans: true })
-				)
+				storage.save(`${id}_background`, background, sharp().jpeg(options))
```

Comment:
Allowing `storage.save` to take a promise as image argument could clear this messy code